test/framework/src/main/java/org/elasticsearch/cluster/project/TestProjectResolvers.java

```diff
@@ -60,7 +61,7 @@ public boolean supportsMultipleProjects() {
      * with the executeOnProject method. This is mostly useful in places where we just need a placeholder to satisfy
      * the constructor signature.
      */
-    public static ProjectResolver singleProjectOnly() {
+    public static ProjectResolver mustExecuteFirst() {
```

Comment:
This method is used in quite a few places in tests. Majority of them (if not all) really just need a projectResolver as a placehoder. The choice of using this variant is probably based on the method name and has no need for its execute behaviour. I renamed this method and added a new `singleProjectOnly` which does only what the name suggests. Hopefully this should satisfy all existing usages. But if not, we can tweak the places where it needs more.